scheduler/scheduler.cpp

```diff
@@ -624,72 +626,104 @@ static string envs_match(CompileServer *cs, const Job *job)
     return string();
 }
 
-static CompileServer *pick_server(Job *job)
+static list<CompileServer *> filter_ineligible_servers(Job *job)
 {
+    list<CompileServer *> eligible;
```

Comment:
I don't see a reason why list should be used over vector here. Any arguments against a vector?

Reply:
Ordinarily, yes, I would prefer a `vector`. However, I was keeping the data type the same during the refactor. I'm also trying to change as little code as possible to get the job done and following the surrounding conventions. Keep in mind that this is C++11 - we're not dealing with "modern" C++ standards here.